compiler-core/src/type_/expression.rs

```diff
@@ -691,6 +691,30 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
                 unify(left.type_(), right.type_())
                     .map_err(|e| convert_unify_error(e, right.location()))?;
 
+                match (&left, &right) {
+                    (TypedExpr::Call { fun, .. }, TypedExpr::Int { value, .. })
+                    | (TypedExpr::Int { value, .. }, TypedExpr::Call { fun, .. }) => {
+                        match fun.as_ref() {
+                            TypedExpr::ModuleSelect {
+                                module_name, label, ..
+                            } => {
+                                let is_list_length_call =
+                                    module_name == "gleam/list" && label == "length";
```

Comment:
Do you think we should also check that this comes from the `gleam_stdlib` package?

Reply:
I think:
- We should also test that it comes form gleam_stdlib. Let's say someone has a todo list and that is streamed or async or something, then calling .length makes sense?
- Once we also test for gleam stdlib we can convert this warning into an error? What would be the drawback?

Reply:
> Do you think we should also check that this comes from the `gleam_stdlib` package?

I took a pass at this in the latest revision.

> Once we also test for gleam stdlib we can convert this warning into an error? What would be the drawback?

One drawback would be that making this an error would be a breaking change.

I think it would be nice to have this as an auto-fixable thing at some point, at which point it probably doesn't matter too much if it's a warning or an error.

A warning seems like a good place to start.

Reply:
> One drawback would be that making this an error would be a breaking change.

I guess this is still technically a breaking change for anyone who:
1. Has usages of `list.length([]) == 0` in their codebase
2. Is using `gleam build --warnings-as-errors` 

I'm new around here, so I don't know what our stance on breaking changes like this is 😄

Reply:
I don't think this will ever be an error, a warning is OK.